### DISCUSSION TITLE
[Meta] Matrix over discord

### DISCUSSION THREAD

**Discussion by Doomsdayrs** [LizardByte/community#29] [Sunshine]
### Describe the Bug

Discord used as primary communication platform

### Expected Behavior

Using open source chat platform

### Additional Context

Matrix is used by both KDE & GNOME for their discussions, it makes sense to have the chats on Matrix as well, as Matrix is now where the open source community is, and is headed.

**Comment by Doomsdayrs** [comment]
If you want, I can setup the matrix space

**Reply by ReenigneArcher** [reply]
Could you share the process of setting this up?

**Reply by Doomsdayrs** [reply]
Open element, View side bar, click the "+" on the left or "create space", create space, create some rooms, set priorities, etc etc

I suggest doing this on desktop, its more fleshed out then mobile

**Reply by ReenigneArcher** [reply]
> Open element, View side bar, click the "+" on the left or "create space", create space, create some rooms, set priorities, etc etc
> 
> I suggest doing this on desktop, its more fleshed out then mobile

Is there any docker version? I don't run my desktop 24/7.

**Reply by sitiom** [reply]
> Could you share the process of setting this up?

[Element](https://element.io/) is the base client for [Matrix](https://matrix.org/); I suggest using it if you are just new to this. The UI is fairly easy to use; you'll learn as you try it. As for the bridging, https://t2bot.io/ provides public instances of the [Discord](https://t2bot.io/discord/) and [Telegram](https://t2bot.io/telegram/) bridge.

**Reply by sitiom** [reply]
> Is there any docker version? I don't run my desktop 24/7.

The web version of Element is available at https://app.element.io/.

**Reply by ReenigneArcher** [reply]
> > Is there any docker version? I don't run my desktop 24/7.
> 
> The web version of Element is available at https://app.element.io/.

So no server is needed?

**Reply by Xiretza** [reply]
Matrix is a federated protocol, you can use any server you want. matrix.org is probably the most commonly used one.

**Reply by Doomsdayrs** [reply]
If you need someone to help you get onto it, many of us are not against it.

**Comment by ReenigneArcher** [comment]
We have discussions here on the github repo.

I have thought about setting up discourse but I don't want to host it and they have limits on the free hosting for open source projects. In the end we're covered here and on discord.

**Comment by ReenigneArcher** [comment]
I looked into this a little bit. Not sure I see a compelling reason to switch.

Try to sell me on it... Why do you want discussions on Matrix? Only because it's open source?

**Comment by Xiretza** [comment]
Just to be clear: Matrix is a chat protocol and as such an alternative to Discord, rather than a forum like Discourse or github Discussions.

Discord is a closed-off centralized service owned and operated by a profit-oriented company, which can and will do whatever they want to increase those profits; including selling off users' data. They can change anything and everything about their platform, and the only recourse users have is to either suck it up or pack their bags and go elsewhere (and they don't make taking your data with you easy). If the company ever dies, everything is gone.

Meanwhile Matrix is a *specification* for a federated chat *protocol*, and is not directly dependent on its client (Element, Fluffychat, weechat-matrix, various bot frameworks, ...) or server (synapse, dendrite, conduit, ...) implementations. Anyone can use whatever client they like best, and either register their account on an existing server, or even host their own server and have it participate in the federated network. A room's state is distributed among all the participating homeservers, so should one server ever go down, only its users are affected.

As mentioned by @Doomsdayrs, Matrix also offers a feature similar to Discord's guilds called Spaces, which are a collection of rooms. Anyone who joins the space can see all the added rooms and join any they find interesting. One possibility that I've seen work well for other projects is to create a Matrix space, inside that create rooms mirroring the discord channel layout, then bridge all those rooms to their respective discord channels. This avoids the need for a sharp transition.

**Comment by sitiom** [comment]
> One possibility that I've seen work well for other projects is to create a Matrix space, inside that create rooms mirroring the discord channel layout, then bridge all those rooms to their respective discord channels.

+1 for this. Bridges are great. Discord users don't need to "switch" while you are giving a choice to other privacy-oriented users and be able to chat with one another:

![image](https://user-images.githubusercontent.com/56180050/174204935-0ce4b831-c24a-439e-9a2a-2ee1be79d324.png)
![image](https://user-images.githubusercontent.com/56180050/174204986-e07b3246-8275-4aca-b9c5-8a6ca87f8426.png)

**Comment by ReenigneArcher** [comment]
Unfortunately with the lack of support for slash commands, threads, buttons, etc. I won't be bridging our discord server to matrix. Slash commands have been a fairly big part of supporting RetroArcher in the past, and now will be with Sunshine as well. Additionally, forum channels are being rolled out soon which we will probably move to for the support channels. Certainly those are not going to be integrated well with matrix either.

You're all welcome to make a community based chat space if you're so inclined.

**Comment by Xiretza** [comment]
It's disheartening to hear that you're planning on entrenching yourself even further in such a closed and proprietary platform, but I suppose convenience does trump a lot of other ideals.

Would you be willing to link to such a community space e.g. in the README?

**Reply by ReenigneArcher** [reply]
I only have so much time in a day. The more time I spend supporting users, the less time I spend on improving our software.

I can put a link in the docs.

**Reply by sitiom** [reply]
Additionally, are you interested in linking such space/room in a separate category/channel in the discord server so that discord users can still get in touch with matrix users?

**Comment by Doomsdayrs** [comment]
https://matrix.to/#/#sunshine:matrix.org has been creates

@sitiom @Xiretza can you tell me your matrix names? I'll make you admins

**Reply by sitiom** [reply]
It should be the same as my username.

**Reply by Xiretza** [reply]
`@xiretza:xiretza.xyz`.
